habitat-lab/habitat/articulated_agents/articulated_agent_base.py

```diff
@@ -224,8 +224,10 @@ def leg_joint_pos(self, ctrl: List[float]):
 
             joint_positions = self.sim_obj.joint_positions
 
+            mt = self.sim_obj.motion_type
             for i, jidx in enumerate(self.params.leg_joints):
-                self._set_motor_pos(jidx, ctrl[i])
+                if mt == MotionType.DYNAMIC:
+                    self._set_motor_pos(jidx, ctrl[i])
```

Comment:
To confirm, the agent joint position will still update in Kinematic mode? Do we also need to have similar logic for the other articulated objects (like cabinets and fridges)? Or are the motors only instantiated on the robots?

Reply:
> To confirm, the agent joint position will still update in Kinematic mode? 
Yes, the line outside the loop:
```
self.sim_obj.joint_positions = joint_positions
```
Does the kinematic position setting. The conditional line here sets the motor targets. These motors have already been removed for kinematic mode, so it would error out.

> Do we also need to have similar logic for the other articulated objects (like cabinets and fridges)?
The changes in rearrange_sim remove all joint motors for all articulated object instances in kinematic mode. Assuming the scene is already loaded and all the furniture is instantiated, that should remove all joint motors. If that assumption isn't valid we should follow-up.